sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlockBlobClient.java

```diff
@@ -594,12 +748,48 @@ public BlockBlobItem uploadFromUrl(String sourceUrl, boolean overwrite) {
     public Response<BlockBlobItem> uploadFromUrlWithResponse(BlobUploadFromUrlOptions options, Duration timeout,
                                                              Context context) {
         StorageImplUtils.assertNotNull("options", options);
-        Mono<Response<BlockBlobItem>> upload = client.uploadFromUrlWithResponse(options, context);
+        return StorageImplUtils.executeOperation(() -> uploadFromUrlWithResponseSync(options, enableSyncRestProxy(context)),
+            timeout);
+    }
+
+    Response<BlockBlobItem> uploadFromUrlWithResponseSync(BlobUploadFromUrlOptions options, Context context) {
+        StorageImplUtils.assertNotNull("options", options);
```

Comment:
Either this check or the check in `uploadFromUtilWithResponse` can be removed as they are duplicative.